aip/client-libraries/4222.md

```diff
@@ -41,25 +187,26 @@ md := metadata.Pairs("x-goog-request-params",
   url.QueryEscape("parent") + "=" + url.QueryEscape(req.GetParent()))
 ```
 
-Much like URL parameters, if there is more than one key-value pair, the `&`
-character is used as the separator. At runtime, if a named parameter is unset
-on the request message, it **should not** be included in the headers.
+At runtime, if a field with the same name as the named parameter is unset on the
+request message, the key-value pair corresponding to that parameter **must not**
+be included in the routing header. If none of the parameters must be included in
+the routing header, the routing header **must not** be sent.
 
 If the [`google.api.http`][http] annotation contains `additional_bindings`,
-these patterns **should** be parsed for additional request parameters. Fields
-not duplicated in the top-level (or `additional_bindings`) pattern **should**
+these patterns **must** be parsed for additional request parameters. Fields
```

Comment:
This paragraph is a bit confusing - it was already confusing though, IMO. It's not clear to me how multiple bindings should actually be handled - I think we should have a whole subsection for that with examples, but that can come in a later PR.

Reply:
Honestly, to clarify this part it would be more useful to direct the reader to read routing.proto